pkg/controller/common/stackmon/sidecar.go

```diff
@@ -29,6 +29,7 @@ func NewMetricBeatSidecar(
 	baseConfigTemplate string,
 	namer name.Namer,
 	url string,
+	basePath string,
```

Comment:
`NewMetricBeatSidecar(..)` and `buildMetricbeatBaseConfig(..)` start to have a lot of parameters. At some point, it might be reasonable to introduce a struct?

Reply:
I think fundamentally basePath should not be in the common stack monitoring code. But I wanted a quick fix we could include in the release. Ideally one would want a bit more more modular code where we can have application/module specific settings in the specific controllers.